### ISSUE TITLE
[frameit] Improve readability of Editor code and prevent repeating calls to fetch the same config parameters

### ISSUE THREAD

**Issue by funnel20** [fastlane/fastlane#11648]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
While finding a solution for issue #11585 we started an interesting discussion in PR #11624 about readability of the code.
This was triggered by `vertical_frame_padding`. I assumed it was a variable, but @janpio pointed out it's actually a method that retrieves the padding value from the Framefile.
So when using this alleged variable several times in the code, it actually performs this method call multiple times, introducing unnecessary computation time.

I want to:

- Increase the readability of the code
- Reduce computation time

This can be done in 2 ways:

**1 - The easy way in Editor class**

- Add local variables/properties for config parameters. E.g. `vertical_frame_padding` and `horizontal_frame_padding`.
- Combine the existing 3 methods `vertical_frame_padding`, `horizontal_frame_padding` and `scale_padding()` to new method `get_padding_from_config`, which assigns the values to variables `vertical_frame_padding` and `horizontal_frame_padding`.
- New method `get_padding_from_config` is only called once in `def frame!(screenshot)`.

**2 - The hard way in Frameit**
The Editor class many times calls `fetch_config[paramater]` even for the same `parameter`. This Class runs in a loop for every screenshot. So if there are a lot of screenshots (210 in one of our projects) `fetch_config[paramater]` is called thousands of times.
While these few config parameters are static for the complete frameit run.
When other Classes want to access a config parameter they have to implement it again; duplicate of code.

I propose the following ideas:

- Define all required config parameters as read-only properties to a dedicated Class. Like new _Config_ or existing _ConfigParser_. Ideally the properties get the same name as the parameters in the Framefile.
- When that class initialises only once, the parameters are fetched from file.
- Initial values are set when the parameter is not defined. This removes the error-prone situation as this is handled now deep in a non-related class, like setting the default value 0.1 of `font_scale_factor` in Editor.
- Also derived parameters (like both horizontal and vertical padding from parameter `padding`) are defined as properties, since they will be used in classes.
- Every class can use the same and once initialised properties over and over again, without any negative computation time effects.

**Benefits**

- [x] Config parameters are initialised only once. This reduces calculation time.
- [x] Derivatives of parameters (like both horizontal and vertical padding from parameter `padding`) are only parsed once and not per class base.
- [x] Central management of config parameters and their default values.
- [x] Removes a lot of code from Editor class.

Another topic to improve the readability is consistently use syntax to recognise class properties.
Currently the following 3 are used all together:

- None, just the name, e.g. `frame`
- self, e.g. `self.frame`
- @, e.g. `@frame`

Please advise which of the 3 is preferred.

##### Complete output when running fastlane, including the stack trace and command used

[INSERT OUTPUT HERE]

### Environment

Fastlane 2.76.1

**Comment by funnel20** [comment]
@ohayon You really helped me with PR #11124 to improve the Frameit code of the Editor class.
Can you please assist here, as this discussion is about coding style, naming conventions and architecture?

**Comment by janpio** [comment]
Did you look how other parts of fastlane solve this problem? `frameit` can't be the only one doing such things...

**Comment by funnel20** [comment]
I'm not a fan of determining these kind of rules by expecting code. The rules might have been updated meanwhile, and by copycatting from an old file I can still fail to comply the new standard.

I'm really looking to some Ruby generic or Fastlane specific code guidelines. 
Maybe these can be added to https://github.com/fastlane/fastlane/blob/master/YourFirstPR.md?

**Comment by funnel20** [comment]
@fastlane-bot Please keep this issue open.
